### ISSUE TITLE
Use `kebab-case` when making tags

### ISSUE THREAD

**Issue by ajensen1234** [mivanit/dendron-citations#3]
Hi! 
This is just a suggestion to align with some of the recommended practices from the dendron website (https://wiki.dendron.so/notes/8d3c8142-7481-40da-9a5c-69a3d4bab697/).

It would be helpful to make the tag generation into a `lower-kebab-case.`

This is happening in the code at:

```
def process_tag_name(s : str, nodot : bool = True) -> str:
	s_new : str = (
		unicodedata.normalize('NFKD', s)
		.encode('ascii', 'ignore').decode('ascii')
		.replace(' - ', '-')
		.replace(' ', '_')
		.replace('\t', '__')
		.replace('\\', '')
		.replace('\n', '')
		.replace('/', '-')
	)

	if nodot:
		s_new = s_new.replace('.', '-')
	
	return s_new
```

I can probably clone the vault and make a pull request to fix this on my own if you'd like me to. Also happy to help with any other development going on in this space!

**Comment by mivanit** [comment]
Thanks for the suggestion!

Enforcing only the kebab-case would be a breaking change (although this probably only affects me..), so I wanted to avoid it -- instead, I've added a command line option that allows this to be done in [this commit](https://github.com/mivanit/dendron-citations/commit/698a45ee9e91e91683a31f5070a9a5359badeb95)

the relevant code is just as follows:
```python
if GLOBAL_CONFIG['kebab-case-tag-names']:
	s_new = s_new.lower()
```

does this work for your use case?



the global config dict is not ideal, but for a tiny script like this I think it's ok. I also might throw some other options in there as well, and perhaps a way to load option from json or yaml

**Comment by ajensen1234** [comment]
For me, the area where this was affecting it (and not necessarily for performance reasons, but rather for maintaining best practices) was in the tag creation where a tag might be called `tags.Simulated_annealing`. 

So a minor change that could also do something like:

```python
if GLOBAL_CONFIG['kebab-case-tag-names']:
        s_new.replace('_', '-')
	s_new = s_new.lower()
```

Would be awesome!

And, for consistency's sake, if the global config option (ironically) used `underscore_case`, it would be more aligned with what the rest of the global configurations are!

**Comment by mivanit** [comment]
Both of these sound good to me! I've implemented them in [ea2eaded..](https://github.com/mivanit/dendron-citations/commit/ea2eadede14852f6818d2246a968ddf474b2a9d8)
